### ISSUE TITLE
ATO fails with this - Error encountered resolving symbol values statically. Function calls are not supported.

### ISSUE THREAD

**Issue by erickbrenesLT** [tiaguinho/ionic2-mask-directive#7]
@tiaguinho @TalissonJunior 
typescript error
            Error encountered resolving symbol values statically. Function calls are not supported. Consider replacing
            the function or lambda with a reference to an exported function (position 12:42 in the original .ts file),
            resolving symbol ɵ2 in
           /node_modules/ionic2-mask-directive/index.js, resolving
            symbol Ionic2MaskDirective in
            /node_modules/ionic2-mask-directive/index.js, resolving
            symbol Ionic2MaskDirective in
            /node_modules/ionic2-mask-directive/index.js, resolving
            symbol MapMaskModule in
            /src/shared/modules/map-mask.module.ts, resolving symbol
            MapMaskModule in /src/shared/modules/map-mask.module.ts

[09:24:41]  ionic-app-script task: "build"
[09:24:41]  Error: The Angular AoT build failed. See the issues above
Error: The Angular AoT build failed. See the issues above
    at /node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:237:55
    at step (/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:32:23)
    at Object.next (/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:13:53)
    at fulfilled (/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:4:58)
    at <anonymous>
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Dispatcher@0.9.14 ionic:build: `ionic-app-scripts build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the Dispatcher@0.9.14 ionic:build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by okonon** [comment]
I was just about to create an issue about this @TalissonJunior @tiaguinho any advice how to work around it?

**Comment by marioalvial** [comment]
same here

**Comment by erickbrenesLT** [comment]
Seems that missing a metadata.json file is a no no too.

**Comment by erickbrenesLT** [comment]
https://angular.io/guide/aot-compiler#function-calls-not-supported

**Comment by marioalvial** [comment]
so what's next? we have to wait till the author fix it?

**Comment by erickbrenesLT** [comment]
Hopefully - I doubt it would be too much heavy lifting.

**Comment by tiaguinho** [comment]
I'll try to fix this until tomorrow

**Comment by okonon** [comment]
@tiaguinho  thanks a lot for looking into this!

**Comment by tiaguinho** [comment]
@erickbrenesLT The error occurs in an Angular class. Maybe this class isn't exported anymore.
Please tell me what ionic version are you using?

**Comment by erickbrenesLT** [comment]
@tiaguinho - Like Okonon said - thanks for checking into this!
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

local packages:

    @ionic/app-scripts : 3.1.7
    Ionic Framework    : ionic-angular 3.9.2

System:

    Node : v8.9.1
    npm  : 5.6.0
    OS   : macOS Sierra

Misc:

    backend : pro

**Comment by erickbrenesLT** [comment]
@tiaguinho and just in case:
Angular CLI: 1.6.4
Node: 8.9.1
OS: darwin x64
Angular: 5.0.0
... common, compiler, compiler-cli, core, forms, http
... platform-browser, platform-browser-dynamic

@angular/.DS_Store: error
@angular/cli: 1.6.4
@angular/tsc-wrapped: 4.1.3
@angular-devkit/.DS_Store: error
@angular-devkit/build-optimizer: 0.0.35
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.4
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.4.2
webpack: 3.8.1

**Comment by okonon** [comment]
@tiaguinho here is my info: 
```
@ionic/cli-utils  : 1.18.0
    ionic (Ionic CLI) : 3.18.0

local packages:

    @ionic/app-scripts : 3.1.7
    Ionic Framework    : ionic-angular 3.9.2

System:

    Node : v8.6.0
    npm  : 5.3.0
    OS   : macOS High Sierra

Misc:

```

**Comment by erickbrenesLT** [comment]
@tiaguinho Any updates or help I can provide?

**Comment by tiaguinho** [comment]
I'm looking into this but seems to be an AOT problem. 
The compiler shows that the error is on this line of the compiled file:
```javascript
var Ionic2MaskDirective = /** @class */ (function () {
```
I'm trying to figure it out if there is something that I can do manually on the file to help you guys.
If you guys find something, please share with me.

Thanks!

**Comment by okonon** [comment]
@tiaguinho thanks a lot for looking into this. I wonder if typescript version is causing this. 
`tsc -v Version 2.5.3`

**Comment by okonon** [comment]
@tiaguinho anything else we can help with to get this resolved?

**Comment by tiaguinho** [comment]
@okonon I'm sorry for the delay.
I try to use this version for compile and got the same compiled code. Next thing I'll try is fixed the compiled code by my self.
At the end of the day, I think this will be fixed.

Sorry for this is taking so long to be fixed.

**Comment by okonon** [comment]
@tiaguinho no problem :) thanks for your help

**Comment by marioalvial** [comment]
Thanks @tiaguinho

**Comment by erickbrenesLT** [comment]
@tiaguinho - thanks.

**Comment by tiaguinho** [comment]
Apparently, this error occurs because of some statically analysis that Angular CLI makes on the app.
Please, change the Angular CLI to 1.5 and tell me if the error this occurs.

Reference: https://github.com/angular/angular-cli/issues/7225

Thanks.

**Comment by erickbrenesLT** [comment]
@tiaguinho I believe I had 1.5 and it failed before.  I just upgraded my angular (5.5.2) and angular cli (1.6.5) and get a bit more info on the failure:

 npm run ionic:build --prod

> Dispatcher@0.10.2 ionic:build /Users/erickbrenes/Sites/name
> ionic-app-scripts build

[10:08:32]  ionic-app-scripts 3.1.7
[10:08:32]  build prod started ...
[10:08:32]  clean started ...
[10:08:32]  clean finished in 4 ms
[10:08:32]  copy started ...
[10:08:32]  deeplinks started ...
[10:08:32]  deeplinks finished in 215 ms
[10:08:32]  ngc started ...
[10:08:41]  typescript error
            src/shared/modules/map-mask.module.ts(7,5): Error during template compile of 'MapMaskModule' Function
            expressions are not supported in decorators in 'Ionic2MaskDirective' 'Ionic2MaskDirective' references 'ɵ2'
            'ɵ2' contains the error at ionic2-mask-directive/index.js.ts(12,42) Consider changing the function
            expression into an exported function.

[10:08:41]  ionic-app-script task: "build"
[10:08:41]  Error: The Angular AoT build failed. See the issues above
Error: The Angular AoT build failed. See the issues above
    at /Users/erickbrenes/Sites/name/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:237:55
    at step (/Users/erickbrenes/Sites/name/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:32:23)
    at Object.next (/Users/erickbrenes/Sites/name/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:13:53)
    at fulfilled (/Users/erickbrenes/Sites/name/node_modules/@ionic/app-scripts/dist/aot/aot-compiler.js:4:58)
    at <anonymous>
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Dispatcher@0.10.2 ionic:build: `ionic-app-scripts build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the Dispatcher@0.10.2 ionic:build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by erickbrenesLT** [comment]
@tiaguinho Upgrade all dependencies and to ionic-app-scripts 3.1.8 = same error as above.

**Comment by erickbrenesLT** [comment]
@tiaguinho
```
"dependencies": {
    "@agm/core": "1.0.0-beta.2",
    "@angular/common": "5.2.1",
    "@angular/compiler": "5.2.1",
    "@angular/compiler-cli": "5.2.1",
    "@angular/core": "5.2.1",
    "@angular/forms": "5.2.1",
    "@angular/http": "5.2.1",
    "@angular/platform-browser": "5.2.1",
    "@angular/platform-browser-dynamic": "5.2.1",
    "@ionic-native/core": "4.5.2",
    "@ionic-native/splash-screen": "4.5.2",
    "@ionic-native/status-bar": "4.5.2",
    "@ionic/storage": "2.1.3",
    "@ngrx/core": "1.2.0",
    "@ngrx/effects": "5.0.0",
    "@ngrx/store": "5.0.0",
    "@ngrx/store-devtools": "5.0.0",
    "@ngx-translate/core": "9.1.1",
    "@ngx-translate/http-loader": "2.0.1",
    "@stomp/stompjs": "4.0.0",
    "angular2-pipes": "2.0.1",
    "aws-sdk": "2.186.0",
    "crypto-js": "3.1.8",
    "electron-log": "2.2.14",
    "electron-reload": "1.2.2",
    "electron-updater": "2.20.1",
    "ionic-angular": "3.9.2",
    "ionic-img-viewer": "2.9.0",
    "ionic2-mask-directive": "1.0.6",
    "ionicons": "3.0.0",
    "jquery": "3.3.1",
    "ngrx-store-ionic-storage": "4.2.0",
    "ngrx-store-logger": "0.2.0",
    "ngx-electron": "1.0.4",
    "ngx-pipes": "2.1.0",
    "rxjs": "5.5.6",
    "sockjs-client": "1.1.4",
    "sw-toolbox": "3.6.0",
    "thenby": "1.2.3",
    "ua-parser-js": "0.7.17",
    "zone.js": "0.8.20"
  },
  "devDependencies": {
    "@angular/cli": "1.6.5",
    "@ionic/app-scripts": "3.1.8",
    "@types/googlemaps": "3.30.5",
    "@types/jquery": "3.3.0",
    "electron": "1.7.11",
    "electron-builder": "19.55.2",
    "minimist": "1.2.0",
    "nodemailer": "4.4.2",
    "npm-run-all": "4.1.2",
    "replace-in-file": "3.1.0",
    "request": "2.83.0",
    "s3": "4.4.0",
    "shelljs": "0.8.1",
    "simple-git": "1.89.0",
    "typescript": "2.6.2"
  },
```

**Comment by tiaguinho** [comment]
@erickbrenesLT thanks for the update. I think I can resolve this until tomorrow with this new informations.

**Comment by eduardolopesUFJF** [comment]
Hello @tiaguinho, I had the same problem here. Did you find some way to fix it? Thanks a lot!

**Comment by tiaguinho** [comment]
I just update the package to version 1.0.7.
Please anyone can test and give me a feedback?

Thanks.

**Comment by eduardolopesUFJF** [comment]
@tiaguinho Wich package I have to update?

**Comment by eduardolopesUFJF** [comment]
I did the update, now I have the following error when execute ionic cordova build android --prod 

 Metadata version mismatch for module C:/Arquivos/Aplicativos/caronAdmin/Angular/node_modules/ionic2-mask-directive/node_modules/@angular/core/core.d.ts, found version 4, expected 3

**Comment by erickbrenesLT** [comment]
@tiaguinho works for me.  Thank you for the time and fixing the issue!  For my own knowledge what was the issue?

**Comment by erickbrenesLT** [comment]
Fixed for Mac - Ionic 3.9.2 - Angular 5.2.1 - Ionic2-mask-directive - 1.0.7

**Comment by sudheerj** [comment]
@tiaguinho Can you please let me know how you fixed this issue. I'm also facing the same issue in one of my library with angular module expression.

`primeng-extensions/components/wizard/wizard.js.ts(129,35): Error during template compile of 'ɵ2'
  Function expressions are not supported in decorators`

And it show at
`var WizardModule = /** @class */ (function () {`

**Comment by ohenriquemonte** [comment]
I'm having the same problem executing the build command with --prod parameter. 
Here is the information:

**ionic cordova build android --prod**

_typescript error
    Error encountered resolving symbol values statically. Calling function 'ɵmakeDecorator', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol Directive in
            .../node_modules/ionic2-mask-directive/node_modules/@angular/core/core.d.ts,
            resolving symbol Ionic2MaskDirective in
            .../node_modules/ionic2-mask-directive/ionic2-mask-directive.d.ts,
            resolving symbol Ionic2MaskDirective in
            .../node_modules/ionic2-mask-directive/ionic2-mask-directive.d.ts
Error: The Angular AoT build failed. See the issues above_


**ionic info**
_[INFO] The Ionic CLI has an update available (3.9.2 => 5.1.0)!
[WARN] No write permissions for global node_modules--automatic CLI updates are disabled.
       To fix, see https://docs.npmjs.com/getting-started/fixing-npm-permissions
       Or, install the CLI update manually:
       npm install -g ionic@latest
cli packages: (/usr/local/lib/node_modules)
    @ionic/cli-utils  : 1.9.2
    ionic (Ionic CLI) : 3.9.2
global packages:
    Cordova CLI : 8.1.2 (cordova-lib@8.1.1)
local packages:
    @ionic/app-scripts : 3.2.3
    Cordova Platforms  : android 7.1.4 ios 4.5.5
    Ionic Framework    : ionic-angular 3.9.2
System:
    Android SDK Tools : 26.1.1
    ios-deploy        : 1.9.4
    Node              : v8.11.2
    npm               : 5.6.0
    OS                : macOS High Sierra_

**Package.json**
_"ionic2-mask-directive": "^1.0.8"_
